controller/dataplane/controller.go

```diff
@@ -219,6 +222,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	deployment, res, err := deploymentBuilder.BuildAndDeploy(ctx, dataplane, r.EnforceConfig, r.ValidateDataPlaneImage)
 	if err != nil {
+		// If Konnect cert Secret isn't ready yet, requeue instead of logging an error stack.
+		if errors.Is(err, certificates.ErrKonnectSecretMissing) {
+			log.Debug(logger, "Konnect certificate Secret not ready yet; requeuing", "dataplane", dpNn)
```

Comment:
The logger that we get [from the context](https://github.com/Kong/kong-operator/blob/5cf7bf832e1cfc2535b1c754c4453438511e5851/controller/dataplane/controller.go#L71) and pass around should already contain dataplane's NN

```suggestion
			log.Debug(logger, "Konnect certificate Secret not ready yet; requeuing")
```